eng/pipelines/coreclr/templates/build-perf-maui-apps.yml

```diff
@@ -156,8 +159,15 @@ steps:
 
   - script: |
       chmod -R a+r .
-      ../dotnet build -bl:MauiiOS.binlog -f net6.0-ios -c Release
-      mv ./bin/Release/net6.0-ios/iossimulator-x64/MauiTesting.app ./MauiiOSDefault.app
+
+      # remove net6.0-maccatalyst to work around https://github.com/dotnet/sdk/issues/21877
+      cp MauiTesting.csproj MauiTesting.csproj.bak
+      sed -i'' -e 's/net6.0-ios;net6.0-maccatalyst/net6.0-ios/g' MauiTesting.csproj
+
+      ../dotnet publish -bl:MauiiOS.binlog -f net6.0-ios --self-contained -r ios-arm64 -c Release /p:_RequireCodeSigning=false
```

Comment:
I'd prefer not to use `dotnet build` with some arbitrary properties set since we should really be testing publish here. We can remove the workaround once we have an SDK with the fix.

Reply:
On the Android side, I wouldn't even recommend using `publish`. It basically just copies extra files.